integration/test_storage.py

```diff
@@ -173,3 +173,61 @@ def test_patch(self):
         # as we're not using ZFS (and can't in these integration tests) we
         # can't really patch anything on a dir volume.
         pass
+
+class TestStorageVolumeSnapshot(StorageTestCase):
+    """Tests for :py:class:`pylxd.models.storage_pool.StorageVolumeSnapshot"""
+
+    # note create and delete are tested in every method
+
+    def create_storage_volume_snapshot(self, pool, volume, snapshot):
+        # 'pool' is either a string or a StoragePool object
+        # 'volume' is either a string of StorageVolume object
+        if isinstance(volume, str):
+            if isinstance(pool, str):
+                pool = self.client.storage_pools.get(pool)
+            volume = pool.volumes.get("custom", volume)
+
+        snapshot = volume.snapshots.create(snapshot)
+        return volume
+
+    def delete_storage_volume_snapshot(self, pool, volume, snapshot):
+        # 'pool' is either a string or a StoragePool object
+        # 'volume' is either a string or a StorageVolume object
+        # 'snapshot' is either a string or a StorageVolumeSnapshot object
+        if isinstance(snapshot, str):
+            if isinstance(volume, str):
+                if isinstance(pool, str):
+                    pool = self.client.storage_pools.get(pool)
+                volume = pool.volumes.get("custom", volume)
+            snapshot = volume.get(snapshot)
+
+        snapshot.delete()
+
+    def test_create_and_get_and_delete(self, pool, volume):
+        if isinstance(volume, str):
+            if isinstance(pool, str):
+                pool = self.client.storage_pools.get(pool)
+            volume = pool.volumes.get("custom", volume)
+
+        pool_name = self.create_storage_pool()
+        self.addCleanup(self.delete_storage_pool, pool_name)
+
+        storage_pool = self.client.storage_pools.get(pool_name)
+        volume = self.create_storage_volume(storage_pool)
+        vol_copy = storage_pool.volumes.get("custom", "vol1")
+        self.assertEqual(vol_copy.name, volume.name)
+        volume.delete()
+
+    @unittest.skip("Can't test PUT on volumes as it doesn't make sense yet")
+    def test_put(self):
+        pass
+
+    @unittest.skip("Can't test .save() on volumes yet - doesn't make sense")
```

Comment:
I agree, I was thinking about changing it [here](https://github.com/canonical/pylxd/pull/584/files/1e2c31520bdce79901f2964fd3d4e2f8633f48d5#diff-688e43164e49516d9e289036ad3f202a8fa91edf5a09ef2b7e3f160d7edeeb34L163-L175) too. Also I will change this to say `volume snapshots` instead of `volumes`.